### DISCUSSION TITLE
The idea of ​​sharp internal corners.

### DISCUSSION THREAD

**Discussion by Key-Alex** [OrcaSlicer/OrcaSlicer#6048] [Ideas]
Hello. 
My name is Alexander. I would like to share an idea for an algorithm for acute interior corners. Usually, all internal corners are rounded when printed, and this sometimes interferes with the assembly of several parts, additional attention has to be paid to this when designing the model. But if, when slicing, we implement an algorithm that would allow making internal corners truly sharp, this would solve many issues. I modeled the part so that you could see the main idea. You need to force the slicer to draw the line a little inward, and then go back. It would be possible to add filters for the angle at which this would need to be done. I don’t know if such an idea already exists, but I decided to propose it. Unfortunately, I am not a programmer and cannot implement this idea on my own.

UPD As an addition, you can hide the seam in these corners. Those. It is possible to force the seam to be made inside the corner (if available).

![corners](https://github.com/SoftFever/OrcaSlicer/assets/76132987/dde06bad-8a6a-45f7-be3b-cb81cfb27a96)

**Comment by CraigHutchinson** [comment]
I have a WIP hexagonal wall storage OpenSCAD model that is found the same issue and implemented the same fix!

In reality this approach just 'hacks' the slicer so ideally a specific approach would be "slow down for corners" and provide an angle/radius threshold as I was surprised to see only overhangs seem to currently slow the wall speed (I'm using BAMBU though)

EDIT: this isn't just for internal corners either, an N-gon can end up fairly round external corners when the wall speed is too high to make the corners!

The issue may only show up when there are lots to print btw. When printing small or singular prints the slow down for cooling will often resolve this.
